bin/generate-gutenboarding-design-thumbnails.mjs

```diff
@@ -12,19 +12,24 @@
  *
  * GENERATE THUMBNAILS:
  *
- * `node ./bin/generate-gutenboarding-design-thumbnails.js`
+ * `node ./bin/generate-gutenboarding-design-thumbnails.mjs`
  *
  * We should bump the version query param on the image src to cache bust the images too in client/landing/gutenboarding/available-designs.ts
  *
  * To check the results, take a look at http://calypso.localhost:3000/new/design and the output folder static/images
  *
  */
 
-const captureWebsite = require( 'capture-website' );
-const sharp = require( 'sharp' );
-const wpUrl = require( '@wordpress/url' );
-const designs = require( '../client/landing/gutenboarding/available-designs-config.json' );
-const screenshotsPath = './static/images/design-screenshots'; // Folder to store output images
+/**
+ * External dependencies
+ */
+import captureWebsite from 'capture-website';
+import sharp from 'sharp';
+import wpUrl from '@wordpress/url';
+import { availableDesignsConfig, DESIGN_IMAGE_FOLDER } from '@automattic/design-picker';
```

Comment:
Since the `available-designs-config.json` file has been moved to the `@automattic/design-picker` package, we need to update the imports.

But it seems that Calypso packages don't expose a `cjs` build, and so:
  - we updated this file to use ES modules syntax, and changed the file extension to `.mjs`
  - we [added a step in `bin/install-gutenboarding-design-thumbnails-dependencies.js`](https://github.com/Automattic/wp-calypso/pull/51306/files#r601641351) to make sure that the `@automattic/design-picker` component is built

At the moment, when running `node ./bin/generate-gutenboarding-design-thumbnails.mjs` on my machine, I get the following error:

```
/Users/marcociampini/Code/wp-calypso/node_modules/@wordpress/components/build/scroll-lock/index.js:41
      htmlDocument = _ref$htmlDocument === void 0 ? document : _ref$htmlDocument,
                                                    ^

ReferenceError: document is not defined
    at createScrollLockComponent (/Users/marcociampini/Code/wp-calypso/node_modules/@wordpress/components/build/scroll-lock/index.js:41:53)
    at Object.<anonymous> (/Users/marcociampini/Code/wp-calypso/node_modules/@wordpress/components/build/scroll-lock/index.js:153:16)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/marcociampini/Code/wp-calypso/node_modules/@wordpress/components/build/popover/index.js:36:42)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
```

Reply:
Update: I removed these changes in favour of a simpler (but potentially more hacky) solution across these commits:
- https://github.com/Automattic/wp-calypso/pull/51306/commits/d53f83e4a64a2de1bbe92abdd613f5c5424280fd
- https://github.com/Automattic/wp-calypso/pull/51306/commits/66c4677e0c3c52374e3d1b51bfd316eb5954bb3d
- https://github.com/Automattic/wp-calypso/pull/51306/commits/4d8da2a80444c5f8f263e52a1dcb6d1761fc6d36

Reply:
> But it seems that Calypso packages don't expose a `cjs` build

That's interesting, they have `tsconfig-cjs.json` that should be producing a `cjs` build. Maybe that doesn't happen automatically and is only there just in case we want to publish to npm.

I think using a relative path to `available-designs-config.json` is fine since it's in a monorepo and this is just a script that's always supposed to be run from within the monorepo.

Reply:
Thanks for the heads-up. When the script only runs within the monorepo, a relative path should be fine!